### PR TITLE
Add AgriFieldNet datamodule

### PR DESCRIPTION
**Pull request by yichiac**
This PR adds the AgriFieldNet datamodule. The training uses satellite images included in the original dataset. We can experiment more with the Sentinel-2 raw scenes.

### CODE REVIEW

**Comment by calebrob6** [comment]
This LGTM!

File: torchgeo/datamodules/agrifieldnet.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            AgriFieldNet,
+            batch_size=batch_size,
+            patch_size=patch_size,
+            length=length,
+            num_workers=num_workers,
+            **kwargs,
+        )
+
+        self.train_aug = AugmentationSequential(
+            K.Normalize(mean=self.mean, std=self.std),
```
Comment (adamjstewart) [inline torchgeo/datamodules/agrifieldnet.py:52]: Just make sure the default values of mean/std are correct for this dataset

File: torchgeo/datamodules/agrifieldnet.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    def setup(self, stage: str) -> None:
+        """Set up datasets.
+
+        Args:
+            stage: Either 'fit', 'validate', 'test', or 'predict'.
+        """
+        dataset = AgriFieldNet(**self.kwargs)
+        generator = torch.Generator().manual_seed(0)
+        (self.train_dataset, self.val_dataset, self.test_dataset) = (
+            random_bbox_assignment(dataset, [0.6, 0.2, 0.2], generator)
```
Comment (adamjstewart) [inline torchgeo/datamodules/agrifieldnet.py:68]: This has a different percentage than CDL. Why?

**Review by adamjstewart — APPROVED** [review]
Couple remaining questions to address during our meeting

**Review by adamjstewart — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**